contracts/smart-wallet/non-upgradeable/Account.sol

```diff
@@ -205,6 +209,18 @@ contract Account is
                         Internal functions
     //////////////////////////////////////////////////////////////*/
 
+    /// @dev Registers the account on the factory if it hasn't been registered yet.
+    function _registerOnFactory() internal virtual {
+        AccountFactory factoryContract = AccountFactory(factory);
+        AccountStorage.Data storage data = AccountStorage.accountStorage();
+        if (!data.isRegistered) {
+            if (!factoryContract.isRegistered(address(this))) {
+                factoryContract.onRegister();
+            }
+            data.isRegistered = true;
```

Comment:
why do you need the account storage flag? isn't it sufficient to check the factory ? is it to save some gas?